Zend/zend_language_scanner.l

```diff
@@ -1110,8 +1110,10 @@ skip_escape_conversion:
 
 static const char *next_newline(const char *str, const char *end, size_t *newline_len) {
 	for (; str < end; str++) {
-		if (*str == '\r') {
-			*newline_len = str + 1 < end && *(str + 1) == '\n' ? 2 : 1;
+		zend_bool is_crlf = *str == '\r' && str + 1 < end && *(str + 1) == '\n';
+		if (is_crlf) {
+			*newline_len = 2;
+			return str;
```

Comment:
It seems to me that this `return str;` is the actual bug fix, and I'd leave it at this. As it is now, a single `\r` would no longer be considered a linebreak (although that might be irrelevant).

Reply:
Yeah, PHP still support `\r`-only linebreaks (for whatever reason...), so I agree that only adding the `return str` would be better.

Reply:
I guess supporting `\r` only lines can't hurt. I updated the PR.